algebra/group/subgroup.lean

```diff
@@ -0,0 +1,148 @@
+import data.set.basic init.function data.equiv init.logic 
+import algebra.group
+
+open set
+
+universes u v
+variables {G : Type u} {H : Type v}
+
+class subgroup [group G] (S : set G) : Prop := 
+    (mul_mem : ∀ {a b}, a ∈ S → b ∈ S → a * b ∈ S) 
+    (one_mem : (1 : G) ∈ S)
+    (inv_mem : ∀ {a}, a ∈ S → a⁻¹ ∈ S) 
+
+class normal_subgroup [group G] (S : set G) extends subgroup S : Prop :=
+    (normal : ∀ n ∈ S, ∀ g : G, g * n * g⁻¹ ∈ S)
+             
+namespace subgroup
+variable [group G]
+
+attribute [simp] subgroup.one_mem 
+                 subgroup.inv_mem 
+                 subgroup.mul_mem
+                 normal_subgroup.normal
```

Comment:
I don't think these should be simp lemmas, simp isn't really designed for this sort of thing. If it helps here, use `local attribute [simp]` instead.

Reply:
I've made this `local` for now, but could just inline them into the `simp`s that use them if that's preferred.